### ISSUE TITLE
question about losses.dice_loss

### ISSUE THREAD

**Issue by XIANYUNYEHE-DEL** [HiLab-git/DTC#18]
Hello，
thanks for your nice work.
I have some confusion as I look your code in losses.dice_loss.
i find dice in the end is the sum. when 1 - loss(why not n - loss), the results is a negative number.
I did an experiment as follows.
```
def dice_loss(score, target):
    target = target.float()
    smooth = 1e-5
    intersect = torch.sum(score * target)
    y_sum = torch.sum(target * target)
    z_sum = torch.sum(score * score)
    loss = (2 * intersect + smooth) / (z_sum + y_sum + smooth)
    loss = 1 - loss
    return loss

a=np.array([[[[[0, 1], [1, 0]],[[0, 1],[1, 0]]]], [[[[0, 1], [1, 0]],[[0, 1],[1, 0]]]]])
b=np.array([[[[[0, 1], [1, 1]],[[0, 1],[1, 1]]]], [[[[0, 1], [1, 1]],[[0, 1],[1, 1]]]]])

a = torch.autograd.Variable(torch.from_numpy(a))
b = torch.autograd.Variable(torch.from_numpy(b))

t = dice_loss(a[:2, 0, :, :, :], b[:2] == 1)
print(t)
```
tensor(-0.6000)

Thats so weird. Did I misunderstand something in any process. Please tell me.
Best.

**Comment by Luoxd1996** [comment]
Hi,
I don't know your met problem. I tested the toy example.
**
def dice_loss(score, target):
    target = target.float()
    smooth = 1e-5
    intersect = torch.sum(score * target)
    y_sum = torch.sum(target * target)
    z_sum = torch.sum(score * score)
    loss = (2 * intersect + smooth) / (z_sum + y_sum + smooth)
    loss = 1 - loss
    return loss

a=np.array([[[[[0, 1], [1, 0]],[[0, 1],[1, 0]]]], [[[[0, 1], [1, 0]],[[0, 1],[1, 0]]]]])
b=np.array([[[[[0, 1], [1, 1]],[[0, 1],[1, 1]]]], [[[[0, 1], [1, 1]],[[0, 1],[1, 1]]]]])

a = torch.from_numpy(a)
b = torch.from_numpy(b)

t = dice_loss(a, b)
print(t)**
The result is :tensor(0.2000);
So I think this function is correct, and it is also widely used in many projects. I think you need to check your inputs carefully.

**Comment by XIANYUNYEHE-DEL** [comment]
thank you for your reply. I think I find my mistake.
